cooked-validators/src/Cooked/MockChain/Monad.hs

```diff
@@ -229,17 +229,22 @@ waitNMilliSeconds n = do
 
 -- ** Modalities
 
--- | Monads supporting modifying transaction skeletons with modalities.
+-- | Monads supporting modifying a certain type of actions with modalities. The 'somewhere'
+-- and 'everywhere' functions receive an argument of type @Action m -> Maybe (Action m)@ because
+-- we want (a) all branches of @somewhere f tree@ to have a guarantee that they had exactly one action
+-- modified by @f@ and (b) we want 'everywhere' to be the dual of 'somewhere', so its type must be the same.
 class (Monad m) => MonadModal m where
-  -- | Applies a modification to all possible transactions in a tree. If a modification
+  type Action m :: *
```

Comment:
I'd probably suggest using `:: Type` here (and below for `Staged`) for some future-proofing (`*`-as-kind will be deprecated eventually AFAIK).

Reply:
Makes sense! Writing `*` is muscle memory at this point though